indico/modules/events/contributions/util.py

```diff
@@ -192,3 +193,53 @@ def generate_spreadsheet_from_contributions(contributions):
             contrib_data['Materials'] = ', '.join(attachments)
         rows.append(contrib_data)
     return headers, rows
+
+
+def get_contributions_with_user_as_submitter(event, user):
+    """Get a list of contributions in which the `user` has submission rights"""
+    contribs = (Contribution.query.with_parent(event)
+                .options(joinedload('acl_entries'))
+                .filter(Contribution.acl_entries.any(ContributionPrincipal.has_management_role('submit')),
+                        ~Contribution.is_deleted)
```

Comment:
excluding deleted contribs is already done in the relationship (which you use due to the `with_parent`)